server/@types/approved-premises/index.d.ts

```diff
@@ -20,7 +21,16 @@ declare module 'approved-premises' {
     nonArrival: Omit<NonArrival, 'id' | 'bookingId'>
   }
 
-  export type DepartureDto = ObjectWithDateParts<'dateTime'> & { departure: Omit<Departure, 'id' | 'bookingId'> }
+  export type DepartureDto = Omit<
+    Departure,
+    'id' | 'bookingId' | 'reason' | 'moveOnCategory' | 'destinationProvider' | 'destinationAp'
+  > &
+    ObjectWithDateParts<'dateTime'> & {
```

Comment:
I'm not clear what `ObjectWithDateParts` is defining even after looking at it's definition?

Reply:
I'll add some comments to it. It's basically defining the shape of an object that has date parts coming from the GDS Date Input (so `${date}-day`, `${date}-month`, ${date}-year`)  I'm not convinced how useful it in in DTOs actually, as we should have converted them by the point that the request body gets converted to a Dto object. I'll tidy this up in a future PR.